routing/record/validation.go

```diff
@@ -100,3 +112,18 @@ func CheckRecordSig(r *pb.Record, pk ci.PubKey) error {
 	}
 	return nil
 }
+
+type RecordSorter []*pb.Record
+
+func (rs RecordSorter) Less(i, j int) bool {
+	mv, _ := MoreValid(rs[j], rs[i])
```

Comment:
As described in https://github.com/ipfs/go-ipfs/pull/999#issuecomment-88603721 -- here may be a good place to defer to user defined functions. ideally we want something like:

```go
best := dht.Validator.MostValid(vals)
```